xayn-ai/src/coi/point.rs

```diff
@@ -121,6 +152,15 @@ pub(crate) trait CoiPoint {
 
     fn set_point(&mut self, embedding: Embedding);
 
+    fn key_phrases(&self) -> &[KeyPhrase] {
+        &[]
+    }
+
+    fn swap_key_phrases(&mut self, candidates: Vec<KeyPhrase>) -> Vec<KeyPhrase> {
+        #![allow(unused_variables)]
+        Vec::new()
+    }
+
```

Comment:
I think it's better to not make this  default impls. 

Given that the new method explicitly accepts `key_phrases` this feels rather wrong.

Reply:
i agree. runtime checking this via using `unimplemented!()` in the default impl would be one way, i think having it compile time checked is even better, so i have split up the `CoiPoint` into additional traits in the same fashion as `CoiPointMerge` before. since the same argument applies for the stats as well, i've splitted out `CoiPointKeyPhrases` and `CoiPointStats` and only implemented them for the `PositiveCoi` and `NegativeCoi`. if we want to avoid the impls for `NegativeCoi` as well, we would have to split the methods of the `CoiSystem`, which we might consider later on but which is out of scope for now.